src/common/libflux/message.c

```diff
@@ -1021,25 +1021,21 @@ int flux_msg_delete_route (flux_msg_t *msg)
 }
 
 /* replaces flux_msg_nexthop */
-int flux_msg_get_route_last (const flux_msg_t *msg, char **id)
+const char *flux_msg_get_route_last (const flux_msg_t *msg)
 {
     struct route_id *r;
 
-    if (!msg || !id) {
+    if (!msg) {
         errno = EINVAL;
-        return -1;
+        return NULL;
     }
     if (!(msg->flags & FLUX_MSGFLAG_ROUTE)) {
         errno = EPROTO;
-        return -1;
-    }
-    if ((r = list_top (&msg->routes, struct route_id, route_id_node))) {
-        if (!((*id) = strdup (r->id)))
-            return -1;
+        return NULL;
     }
-    else
-        (*id) = NULL;
-    return 0;
+    if ((r = list_top (&msg->routes, struct route_id, route_id_node)))
+        return r->id;
+    return NULL;
 }
```

Comment:
Problem: returns NULL on both error (with errno set) and if `list_top()` returns NULL (without errno set).

I think we talked about this but it didn't immediately pop out when I scanned the discussion in this pr - should this function and  other route accessors dispense with trying to set errno altogether?

Reply:
i actually went back and forth on the setting errno.  I feel we shouldn't set errno, but I ultimately decided to set errno simply b/c I saw other parts of flux-core do it when the API style was similar.  We're probably not super consistent about how we do this.

Since you mention it, lets go back to not setting errno, since I think that makes more sense.